Mathlib/Tactic/NoncommRing.lean

```diff
@@ -53,8 +60,13 @@ macro_rules
           -- Pull `zsmul n` out the front so `abel` can see them.
           mul_smul_comm, smul_mul_assoc,
           -- Pull out negations.
-          neg_mul, mul_neg] |
+          neg_mul, mul_neg,
+          -- user-specified simp lemmas
+          $rules',*] |
```

Comment:
Putting these as part of the same `simp` call in which all the other lemmas are called could have some unwanted side effects. For example, this fails because we also call `mul_add`:
```
example (h : a * (b + c) = 0) : a * (b + c) = 0 := by
  noncomm_ring [h] -- fails
```
In noncommutative rings, you're much more likely (than in the commutative setting) to have equalities like `h` which would be otherwise simplified by the simp lemmas in `noncomm_ring`.

EDIT: I realized that the solution to this is actually not on the part of `noncomm_ring`, but rather on the user to write `noncomm_ring [↓h]`, which works. But maybe we could add this to the tactic's documentation, as I suspect most people aren't aware of `↓` yet.

Reply:
I think it's reasonable for your example to fail. If you want to rewrite with `h` first we already can do that: `rw [h]; noncomm_ring`. The syntax `noncomm_ring [h]` is for cases where you  want to normalize, then rewrite, and perhaps then normalize further.

Reply:
The proof that prompted me to implement this was something like `noncomm_ring; rw [h1, h2]; noncomm_ring; rw [h1]; noncomm_ring` and it felt tedious.